### ISSUE TITLE
observable.ref rerendering

### ISSUE THREAD

**Issue by user753** [mobxjs/mobx-react-lite#207]
```
export class MobxStore {
  @observable
  state = {value: 0, value2: 0}

  increase(){
    this.state.value += 1
  }

  increase2(){
    this.state.value2 += 1
  }
}

export const MobxCounter = (props: {store: MobxStore}) => {
  console.log("render")
  return useObserver(() =>
      <div>Counter {props.store.state.value}
        <button onClick={() => props.store.increase()}>increase</button>
        <button onClick={() => props.store.increase2()}>increase2</button>
      </div>
    )
}

```
When I press increase2 button component doesn't rerender as expected.
But if I use `observable.ref`
```
export class MobxStore {
  @observable.ref
  state = {value: 0, value2: 0}

  increase(){
    this.state = produce(this.state, draft => {
      draft.value += 1
    })
  }

  increase2(){
    this.state = produce(this.state, draft => {
      draft.value2 += 1
    })
  }
}
```
and press increase2 it starts to rerender.
Is there any way to use `observable.ref` and somehow prevent this rerendering?

**Comment by FredyC** [comment]
Please setup CodeSandbox to show the problem in action.

**Comment by user753** [comment]
@FredyC https://codesandbox.io/s/friendly-mobx-k7vcl
if you press increase2 `MobxCounter` will be rerendered, but it doesn't use `value2`

**Comment by FredyC** [comment]
Thanks for the sandbox. It's truly interesting. Unfortunately, I have no idea what's wrong there. Hopefully, someone else can chime in.

**Comment by mweststrate** [comment]
@user753 seems to work as it should here. In the orignal example, it updates value2, which isn't used in render, so it doesn't need to re-render, that optimization is kinda the whole point of mobx. 

In your second example, you are mixing two paradigms; you produce an entire new state using immer, and than storing a reference to that new state using mobx observables. That works correctly, but is horribly overcompicating things imho. Just pick one of the paradigms and use it consistently. Mixing the two might just make things unnecessarily complicated. 

In other words, in both cases things are working as expected.

**Comment by FredyC** [comment]
@mweststrate I think he was more concerned that it does re-render when `value2` changes without it being used in render.

**Comment by mweststrate** [comment]
That is correct though, re-render does not happen  because `value2` changed, it happens becouse `state` changed to a completely new object. MobX does change propagation, not change _detection_ (which would required a fully recursive comparison).

If the two paradigms wouldn't be mixed up, this wouldn't happen, as in the mutable paradigm no reason why state would need to be replaced here.

**Comment by user753** [comment]
@mweststrate 
>Mixing the two might just make things unnecessarily complicated.
I understand that. I like the `mobx` approach with proxies but I would also prefer to use immutability.
Is there any way to use `mobx` in such way? 
There is a package https://github.com/theKashey/proxyequal It's like the opposite of `immer`, for detecting used keys. Would it be possible to use `mobx` in similar way, something like readonly mode?

**Comment by FredyC** [comment]
@user753 Why do you want to mix both? MobX is based on mutability. You really have to choose one or another. From your example you are not even using any reactivity of MobX, so why do you need it?

**Comment by mweststrate** [comment]
I like spaghetti and ice cream. But mixing them is a worse experience than
either of them separately :)

**Comment by jeremy-coleman** [comment]
https://i2.wp.com/www.thekitchenmaus.com/wp-content/uploads/2016/06/spaghetti_ice_cream_6443.jpg?resize=700%2C466
